src/backend/gl/src/queue.rs

```diff
@@ -386,6 +386,22 @@ impl CommandQueue {
                     error!("Instanced indexed drawing is not supported");
                 }
             }
+            com::Command::DrawIndirect {
+                primitive,
+                offset,
+                draw_count
+            } => {
+                let gl = &self.share.context;
+
+                unsafe {
+                    gl.DrawArraysInstanced(
```

Comment:
That's not an indirect draw call. Also the `draw_count` parameter seems to be misused and an instance count of 0 appears a bit suspicious to me.

Reply:
@msiglreith Looking at this documentation https://www.khronos.org/opengl/wiki/Vertex_Rendering#Indirect_rendering it sounded like DrawIndirect would call glDrawArraysInstancedBaseInstance underneath, which I thought was the same as glDrawArraysInstanced with 0 as the instance.

Reply:
Indirect drawing gets its information from a GPU buffer though; specifically, the one bound to the  `GL_DRAW_INDIRECT_BUFFER` binding, which `DrawArraysInstanced` does not. `draw_count` is only a parameter for `glMultiDrawArraysIndirect` and it is telling it how many successive indirect draw call information structures to look for after `offset` into the `GL_DRAW_INDIRECT_BUFFER`. The `count` parameter in `DrawArraysInstanced` that you are passing it into is telling the call how may *indices* to render in the currently bound vertex buffers, not how many draw calls to make.

Reply:
I removed this code because I'm not actually doing any indirect draws. I was also doing it terribly wrong :P